core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -67,24 +68,10 @@ public GenericManifestFile(Schema avroSchema) {
     this.avroSchema = avroSchema;
   }
 
-  GenericManifestFile(InputFile file, int specId) {
```

Comment:
The removed constructor is not used anymore and is package private, so should be safe to remove entirely.